adapters/repos/db/lsmkv/metrics.go

```diff
@@ -26,12 +27,19 @@ type (
 	TimeObserver       func(start time.Time)
 )
 
+var compactionDurationBuckets = prometheus.ExponentialBuckets(0.01, 2, 18) // 0.01s → 0.02s → ... → ~163.84s
+
 type Metrics struct {
-	CompactionReplace            *prometheus.GaugeVec
-	CompactionSet                *prometheus.GaugeVec
-	CompactionMap                *prometheus.GaugeVec
-	CompactionRoaringSet         *prometheus.GaugeVec
-	CompactionRoaringSetRange    *prometheus.GaugeVec
+	criticalBucketsOnly bool
```

Comment:
is it still needed? `monitorSegments` method is removed

Reply:
thanks @aliszka for reviewing it. I'm reverting that change and I'll address those two options in a separate PR